From: Steve Wise <swise@opengridcomputing.com.com>
Date: Tue, 12 Apr 2016 06:54:54 -0700
Subject: iw_cxgb4: initialize ibdev.iwcm->ifname for port mapping
Patch-mainline: v4.6-rc6
Git-commit: 851d7b6b8a954b28f85e62a53cbba472907e21e4
References: bsc#963897 FATE#320114

The IWCM uses ibdev.iwcm->ifname for registration with the iwarp
port map daemon.  But iw_cxgb4 did not initialize this field which
causes intermittent registration failures based on the contents of the
uninitialized memory.

Fixes: 170003c894d9 ("iw_cxgb4: remove port mapper related code")
Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/infiniband/hw/cxgb4/provider.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/infiniband/hw/cxgb4/provider.c
+++ b/drivers/infiniband/hw/cxgb4/provider.c
@@ -578,6 +578,8 @@ int c4iw_register_device(struct c4iw_dev
 	dev->ibdev.iwcm->add_ref = c4iw_qp_add_ref;
 	dev->ibdev.iwcm->rem_ref = c4iw_qp_rem_ref;
 	dev->ibdev.iwcm->get_qp = c4iw_get_qp;
+	memcpy(dev->ibdev.iwcm->ifname, dev->rdev.lldi.ports[0]->name,
+	       sizeof(dev->ibdev.iwcm->ifname));
 
 	ret = ib_register_device(&dev->ibdev, NULL);
 	if (ret)
